WordPress/WordPressApi/WordPressComRestApi.swift

```diff
@@ -211,6 +211,14 @@ public class WordPressComRestApi: NSObject
         }
         return !(authToken.isEmpty)
     }
+
+    override public var hashValue: Int {
+        if let validAuthToken = oAuthToken {
+            return validAuthToken.hashValue
+        } else {
+            return "anonymous".hashValue
+        }
+    }
```

Comment:
From the `Hashable` documentation, `x == y implies x.hashValue == y.hashValue.`
To satisfy that, the hash value should take into account all the relevant properties of the object, and this is missing the user agent. I'm not too happy about the "anonymous" magic string.

How about this:

```swift
override public var hashValue: Int {
    return [oAuthToken, userAgent]
        .flatMap({ $0 })
        .joinWithSeparator(",")
        .hashValue
}
```

Or another variant:

```swift
override public var hashValue: Int {
    return "\(oAuthToken),\(userAgent)".hashValue
}
```